lib/gametictac.rb

```diff
@@ -0,0 +1,64 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+class Tictactoe
+  attr_reader :board
+
+  def initialize()
+    @symbol1 = 'X'
+    @symbol2 = 'O'
+    @board = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+    @turn = true
+  end
+
+  def display
+    puts "        #{board[0]} | #{board[1]} | #{board[2]} "
+    puts '      -------------'
+    puts "        #{board[3]} | #{board[4]} | #{board[5]} "
+    puts '      -------------'
+    puts "        #{board[6]} | #{board[7]} | #{board[8]} "
+  end
+
+  def valid_movements?(position)
+    (position.positive? && position < 10) && board[position.to_i - 1] == ' '
+  end
+
+  def move(position)
+    board[position.to_i - 1] = @turn ? @symbol1 : @symbol2
+    @turn = !@turn
+    if @turn == false
+      puts "Last symbol played 'X'\n"
+    else
+      puts "Last symbol played 'O'\n"
+    end
+  end
+
+  def full?
+    board.none? ' '
+  end
+
+  def win?
+    if board[0] == @symbol1 && board[1] == @symbol1 && board[2] == @symbol1 ||
+       board[3] == @symbol1 && board[4] == @symbol1 && board[5] == @symbol1 ||
+       board[6] == @symbol1 && board[7] == @symbol1 && board[8] == @symbol1 ||
+       board[0] == @symbol1 && board[3] == @symbol1 && board[6] == @symbol1 ||
+       board[1] == @symbol1 && board[4] == @symbol1 && board[7] == @symbol1 ||
+       board[2] == @symbol1 && board[5] == @symbol1 && board[8] == @symbol1 ||
+       board[0] == @symbol1 && board[4] == @symbol1 && board[8] == @symbol1 ||
+       board[2] == @symbol1 && board[4] == @symbol1 && board[6] == @symbol1
+      'X won!'
+    elsif board[0] == @symbol2 && board[1] == @symbol2 && board[2] == @symbol2 ||
+          board[3] == @symbol2 && board[4] == @symbol2 && board[5] == @symbol2 ||
+          board[6] == @symbol2 && board[7] == @symbol2 && board[8] == @symbol2 ||
+          board[0] == @symbol2 && board[3] == @symbol2 && board[6] == @symbol2 ||
+          board[1] == @symbol2 && board[4] == @symbol2 && board[7] == @symbol2 ||
+          board[2] == @symbol2 && board[5] == @symbol2 && board[8] == @symbol2 ||
+          board[0] == @symbol2 && board[4] == @symbol2 && board[8] == @symbol2 ||
+          board[2] == @symbol2 && board[4] == @symbol2 && board[6] == @symbol2
```

Comment:
- [ ] [REQUIRED] DRY rule should be respected. Instead of checking the winning positions for X and O separately, create a method and reuse it. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/dry.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)). An example is shown below:
```
  def check_win(player_symbol)
    # for rows
    all_equal(board[0..2], player_symbol) ||
    all_equal(board[3..5], player_symbol) ||
    all_equal(board[6..8], player_symbol)

    # and analogical as above for columns and diagonals
  end

  def all_equal(subarray, player_symbol)
    return true if subarray.uniq.size == 1 && subarray.uniq == player_symbol
  end
```